parser/match_operations.go

```diff
@@ -482,44 +535,55 @@ func comparisonMatch(
 		return fmt.Errorf("%w: operation addresses not equal", err)
 	}
 
-	for _, amountMatch := range descriptions.OppositeAmounts {
-		if len(amountMatch) != oppositesLength { // cannot have opposites without exactly 2
-			return fmt.Errorf("cannot check opposites of %d operations", len(amountMatch))
-		}
+	oppositeAmountMatchingOperations := [][][]int{
+		descriptions.OppositeOrZeroAmounts,
+		descriptions.OppositeAmounts,
+	}
+	oppositeAmountCheckers := []func(*types.Operation, *types.Operation) error{
+		oppositeOrZeroAmounts,
+		oppositeAmounts,
+	}
```

Comment:
The 3d array makes grasping the code kinda difficult (this file is pretty difficult to read as is :( ). 

Can we instead utilize a function to which we pass `amountPairs`, `amountChecker`  and anything else that is needed (`matches`) and call it twice. Once for `OppositeAmounts ` and once for `OppositeOrZeroAmounts `? 🙏